src/App.tsx

```diff
@@ -1,61 +1,125 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App = () => {
+  const getUserById = (id: number, errorMessage = 'User not found'): User => {
+    const user = usersFromServer.find(usr => usr.id === id);
+
+    if (!user) {
+      throw new Error(errorMessage);
+    }
+
+    return user;
+  };
+
+  const [selectedUser, setSelectedUser] = useState<number>(0);
+  const [todos, setTodos] = useState<Todo[]>(() =>
+    todosFromServer.map(todo => ({
+      ...todo,
+      user: getUserById(todo.userId),
+    })),
+  );
+
+  const [title, setTitle] = useState<string>('');
+  const [inputTouched, setInputTouched] = useState<boolean>(false);
+  const [userNotSelected, setUserNotSelected] = useState<boolean>(false);
+
+  const getNextTodoId = (): number => {
+    const maxId =
+      todos.length > 0 ? Math.max(...todos.map(todo => todo.id)) : 0;
+
+    return maxId + 1;
+  };
+
+  const resetForm = (): void => {
+    setTitle('');
+    setInputTouched(false);
+    setSelectedUser(0);
+    setUserNotSelected(false);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const isUserNotSelected = selectedUser === 0;
+    const isTitleEmpty = !title.trim();
+
+    setUserNotSelected(isUserNotSelected);
+    setInputTouched(isTitleEmpty);
+
+    if (isUserNotSelected || isTitleEmpty) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNextTodoId(),
+      title: title.trim(),
+      completed: false,
+      userId: selectedUser,
+      user: getUserById(selectedUser),
+    };
+
+    setTodos(prevTodos => [...prevTodos, newTodo]);
+
+    resetForm();
+  };
```

Comment:
In the `handleSubmit` function, while you are checking for a selected user and a non-empty title, there is no error handling for the `getUserById` function call. If `getUserById` throws an error, it could disrupt the form submission process. Consider adding a try-catch block around the `getUserById` call to handle potential errors gracefully.